.github/actions/setup-venv/action.yml

```diff
@@ -15,6 +15,8 @@ runs:
       uses: actions/setup-python@v4
       with:
         python-version: ${{ inputs.python-version }}
+        cache: "pip"
+        cache-dependency-path: "pyproject.toml"
```

Comment:
Do we still need the caching steps below if we're using the built-in caching mechanism from `setup-python`?

Reply:
Ok turns out this only caches the `pip` cache, which is not great. I've tried doing this before and it's usually a lot slower than caching the virtual env itself.